### PR TITLE
Rename Image-Exporter to Orca

### PR DESCRIPTION
**Pull request by JamesCropcho**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
 
-[![CircleCI](https://circleci.com/gh/plotly/image-exporter.svg?style=svg)](https://circleci.com/gh/plotly/image-exporter)
+[![CircleCI](https://circleci.com/gh/plotly/orca.svg?style=svg)](https://circleci.com/gh/plotly/orca)
 
 This repo contains source code for:
 
-- `plotly-graph-exporter` standalone app,
-- `plotly-image-exporter` npm package, and
+- `orca` standalone app,
+- `@plotly/orca` npm package, and
```
Comment (etpinard) [inline README.md]: just `orca` here not `@plotly/orca`

File: README.md [hunk 2]
```diff
@@ -48,38 +48,38 @@ library(plotly)
 p <- plot_ly(x = 1:10, y = 1:10, color = 1:10)
 b <- plotly_build(p)$x[c("data", "layout")]
 json <- plotly:::to_JSON(b)
-cmd <- sprintf("plotly-graph-exporter '%s' -o r-export-test.png", json)
+cmd <- sprintf("orca '%s' -o r-export-test.png", json)
 system(cmd)
 ```
 
-## `plotly-image-exporter` npm package
+## `@plotly/orca` npm package
```
Comment (etpinard) [inline README.md]: same ...

File: README.md [hunk 3]
```diff
@@ -48,38 +48,38 @@ library(plotly)
...
 
 ### Install
 
 With Node.js (v6.x or v8.x) and npm installed:
 
 ```
-$ npm install -g electron plotly-image-exporter
+$ npm install -g electron @plotly/orca
```
Comment (etpinard) [inline README.md]: one more

File: README.md [hunk 4]
```diff
@@ -1,34 +1,34 @@
-# Plotly Image Exporter
+# Plotly Orca: Open-source Report Creator App
```
Comment (etpinard) [inline README.md]: Not sure we want that _Open-source Report Creator App_ bit in our title. That was more of a joke. I'd remove it for now.

File: README.md (hunk 1 above)
Comment (JamesCropcho) [reply README.md]: I have made this change.

File: README.md (hunk 2 above)
Comment (JamesCropcho) [reply README.md]: I have made this change.

File: README.md (hunk 3 above)
Comment (JamesCropcho) [reply README.md]: I have made this change.

File: README.md (hunk 4 above)
Comment (JamesCropcho) [reply README.md]: I have made this change.

**Comment by JamesCropcho** [comment]
I appear to have fixed the build, albeit by thinking way outside the box, relative to the task-at-hand: https://github.com/plotly/orca/commit/9aefae340488aac8d6767bc60e23c932c50f99d6

**Comment by JamesCropcho** [comment]
@scjody @etpinard 
> So if I understand correctly, docker-build-and-push has been broken since we changed the repo's name to orca.
>
> Now, why does adding libgtk2.0 fix this issue?

Although I do not know, nor even feel like I have a "strong" theory, my hunch is that it relates to some difference in package-json.lock.

@etpinard:
> But, please just get the tests to pass off one of those PRs w/o having to update quay.io things?

This PR meets your above criterion for merging, yes?

**Comment by JamesCropcho** [comment]
@etpinard 

> Can you confirm that Ubuntu versions differ in the built docker container before and after orca? To do so, I would click Rerun job with SSH on CircleCI and run lsb_release -a inside the built docker container.

Actually, I'm pretty sure `lsb_release -a` would look identical in both, i.e. that we are using Xenial. That command's output does not reflect the versions of any of the packages installed (not even the kernel). Ubuntu versions are only incremented exactly once every six months.

> What is libgtk2.0?

It's part of GTK+, which is an old-school *nix toolkit for creating GUIs.

>How did you think that adding it to our Dockerfile would fix the issue?

_They call me the container whisperer…_ No, just kidding. The only thing that the CircleCI logs were telling me was that the smoke test was failing. So I SSH'd into the build box and ran a `docker run` command without the `--detach` or any other daemonizing arguments. Container creation failed, and part of the output of my attempt was

> `error while loading shared libraries: libgtk-x11-2.0.so.0: cannot open shared object file: No such file or directory`

The above is 98% of what led me to my fix. The other 2% was me monkeying around, trying to determine which Aptitude package might have the shared object we needed.

===

All of that make sense? How shall we best proceed towards initial release of Orca?

**Comment by etpinard** [comment]
> Actually, I'm pretty sure lsb_release -a would look identical in both, i.e. that we are using Xenial.

Ok. You're probably right here. But can you try finding a way to confirm your hypothesis? There's a gotta a way to check whether libgtk2.0 is installed or not.

**Comment by etpinard** [comment]
Moreover @JamesCropcho how big is libgtk2.0? We (probably) won't need it for our image server. So perhaps we could remove it from the Dockerfile and install only on CI?

**Comment by JamesCropcho** [comment]
@etpinard 
> Ok. You're probably right here. But can you try finding a way to confirm your hypothesis? There's a gotta a way to check whether libgtk2.0 is installed or not.

Let's say that I do as you propose—I somehow figure out the date of the last uncached build of that Docker Image layer, pull the corrresponding Xenial tag if it still exists, and build a cousin image without libgtk2.0 (we would never be able to get historically accurate Aptitude package versions, however). And let's say that this image works even without that library. But, what then? You already proved (thank you) on your dummy branch, that this fix in isolation differentiated between broken and fixed builds. I don't believe we ware fortunate enough that either hypothetical outcome of the proposed experiment would have any tangible strategic effect for us.

Given the above, would you still opt for me to proceed to see if we have the necessary historical fact(s) for me to perform the experiment?

> Moreover @JamesCropcho how big is libgtk2.0? We (probably) won't need it for our image server. So perhaps we could remove it from the Dockerfile and install it only on CI?

Personally, that sounds scary to me. Wouldn't it defeat the purpose of the smoke test, to modify the Image after running it?

**Comment by etpinard** [comment]
> I somehow figure out the date of the last uncached build of that Docker Image layer, pull the corrresponding Xenial tag if it still exists,

No need to pull down anything. Just pick an old build on CircleCI and click on _rebuild with SSH_ button on the top right corner. 

> Wouldn't it defeat the purpose of the smoke test

If the smoke test requires libgtk2.0 to run, we should be able to reproduce it locally. Have you been able to reproduce those `error while loading shared libraries: libgtk-x11-2.0.so.0: cannot open shared object file: No such file or directory` errors locally? 

> Personally, that sounds scary to me.

Maybe. But adding something in a Dockerfile we no apparent reason except _it gets the test to pass on ci_ is even scarier. 

-----

Oh well, looks like you don't feel like investigating this hiccup further - which I can't really blame you for, it sounds pretty annoying. **But** can at least make a one-commit noise-less PR with just that one patch in the Dockerfile with a thorough explanation of why you did this to get our Docker expert @scjody to review the change? Thank you.

**Comment by JamesCropcho** [comment]
@etpinard 
> Oh well, looks like you don't feel like investigating this hiccup further - which I can't really blame you for, it sounds pretty annoying.

I do admit that it's annoying (ha). However, as explained above my reticence to further explore is not laziness nor feeling annoyed but rather that I question what if any practical change to would result in even a positive result from the proposed experiment.

I hope that makes sense, and that you find it reasonable.

I have created the precise pull request you seek, and requested Jody's (and your) diligent eye for its review.

**Comment by etpinard** [comment]
@JamesCropcho would you be interested in making `orca` a single-entry-point app as discussed in https://github.com/plotly/orca/issues/21#issuecomment-384298071 ?

**Comment by JamesCropcho** [comment]
@etpinard 
> @JamesCropcho would you be interested in making orca a single-entry-point app as discussed in https://github.com/plotly/orca/issues/21#issuecomment-384298071 ?

If am the best person for the job, then yes, I am able to do so. However, I would need to commence on or after Wednesday, 9 May. (This does not mean I am AWOL for my current task(s) and any other communication until then of course!) What do you think of that date?

I knew that task was on the radar. Still, I have focused on making the Orca rename as being earlier in the pipeline, and have also make said renaming independent and separable from converting to single entry-point.

**Comment by etpinard** [comment]
Ok. Thanks for the info.

I'll try to tackle it before May 9th, if not I'll ping you.

File: test/integration/plotly-orca_test.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ const pkg = require('../../package.json')
 const { paths } = require('../common')
 
 const ROOT_PATH = path.join(__dirname, '..', '..')
-const BIN = path.join(ROOT_PATH, 'bin', 'plotly-graph-exporter.js')
+const BIN = path.join(ROOT_PATH, 'bin', 'orca.js')
```
[the remark this answers is not in the document]
Comment (JamesCropcho) [reply test/integration/plotly-orca_test.js]: Okay, I have made this requested change.

File: src/util/pdftops.js [hunk 6]
```diff
@@ -6,7 +6,7 @@ const series = require('run-series')
 const uuid = require('uuid/v4')
 const os = require('os')
 
-const PATH_TO_BUILD = path.join(os.tmpdir(), 'plotly-graph-exporter-build')
+const PATH_TO_BUILD = path.join(os.tmpdir(), 'plotly-orca-build')
```
[the remark this answers is not in the document]
Comment (JamesCropcho) [reply src/util/pdftops.js]: Okay, I have made this requested change.

File: src/util/create-index.js [hunk 7]
```diff
@@ -5,7 +5,7 @@ const isNonEmptyString = require('./is-non-empty-string')
 const os = require('os')
 
 const COMPONENT_GLOBAL = 'PlotlyExporterComponent'
-const PATH_TO_BUILD = path.join(os.tmpdir(), 'plotly-graph-exporter-build')
+const PATH_TO_BUILD = path.join(os.tmpdir(), 'plotly-orca-build')
```
[the remark this answers is not in the document]
Comment (JamesCropcho) [reply src/util/create-index.js]: Okay, I have made this requested change.

**Comment by etpinard** [comment]
Thanks @JamesCropcho !

**Pull request merged**